### DISCUSSION TITLE
Couldn't this be massively optimized via search pages?

### DISCUSSION THREAD

**Discussion by Fmstrat** [jef/streetmerchant#1813] [Ideas and feature requests]
I notice that this tracks each product individually. However, many of these sites have search filters that allow you to see all the prices and all "in-stock" for each item via the search results page. For instance, to see all cards (over $300 to limit results) for stores:

- https://www.officedepot.com/catalog/search.do?N=5&Ntt=geforce&Ntk=all&Ntx=mode+matchpartialmax&Nty=1&cbxRefine=1462159&lowerBound=300&inStock=true
- https://www.bestbuy.com/site/searchpage.jsp?_dyncharset=UTF-8&cp=3&id=pcat17071&iht=y&keys=keys&ks=960&list=n&qp=category_facet%3DGPUs%20%2F%20Video%20Graphics%20Cards~abcat0507002%5Ecurrentprice_facet%3DPrice~350%20to%20Up&sc=Global&st=geforce&type=page&usc=All%20Categories
- https://www.adorama.com/l/?sel=Item-Condition_New-Items|Internal-Interface_PCIe|Price-Range_-dollar-300-to-dollar-2500&searchinfo=geforce
- https://www.zotacstore.com/us/graphics-cards/geforce-rtx-30-series
- https://www.newegg.com/p/pl?quicklink=true&N=100007709%20601357282%208000&PageSize=96
- https://www.microcenter.com/search/search_results.aspx?Ntk=all&sortby=match&N=4294966937+4294808776&myStore=true

Couldn't the DOM be parsed there to determine stock? One hit for all cards.

**EDIT:**
- Added newegg
- Added microcenter

**Comment by Fmstrat** [comment]
Adding some more:
- https://www.bhphotovideo.com/c/products/Graphic-Cards/ci/6567/N/3668461602?sort=PRICE_LOW_TO_HIGH&filters=fct_nvidia-geforce-series_5011%3Ageforce-rtx-3070%7Cgeforce-rtx-3080%7Cgeforce-rtx-3090
- https://www.officedepot.com/a/browse/graphic-cards/N=5+1461934&cbxRefine=1486419&cbxRefine=1486420&cbxRefine=1486421&cbxRefine=1486422&cbxRefine=1486424&cbxRefine=1498445/;jsessionid=0000M4nqne2gzzcq5KA2kMg0ZGF:17h4h7ado
- https://www.adorama.com/l/Computers/Computer-Components?perPage=75&sel=Item-Condition_New-Items&SearchInfo=geforce+rtx
- https://www.pny.com/consumer/shop-pny

That's 10 web calls that would cover every GTX call currently being made (which is a lot).

**Reply by aellch** [reply]
Before I moved to streetmerchant I was using visualping exactly like this.  This feels doable and I really hope it gets implemented.
